internal/telemetry/client.go

```diff
@@ -531,8 +527,8 @@ func (c *client) scheduleSubmit(r *Request) {
 // sending the app-heartbeat event and flushing any outstanding
 // telemetry messages
 func (c *client) backgroundHeartbeat() {
-	c.mu.Lock()
-	defer c.mu.Unlock()
+	c.mu.RLock()
```

Comment:
same here, I think we need to just use a normal Lock since flush modifies the client